src/Util/Crypto.php

```diff
@@ -150,6 +151,48 @@ public static function newKeypair($bits)
 		return $response;
 	}
 
+	/**
+	 * Create a new elliptic curve key pair
+	 *
+	 * @return array with the elements "prvkey", "vapid" and "pubkey"
+	 */
+	public static function newECKeypair()
+	{
+		$openssl_options = [
+			'curve_name'       => 'prime256v1',
+			'private_key_type' => OPENSSL_KEYTYPE_EC
+		];
+
+		$conf = DI::config()->get('system', 'openssl_conf_file');
+		if ($conf) {
+			$openssl_options['config'] = $conf;
+		}
+		$result = openssl_pkey_new($openssl_options);
+
+		if (empty($result)) {
+			Logger::notice('new_keypair: failed');
+			return [];
```

Comment:
This should throw an exception, otherwise `Model\Subscription::getVapidKey` will return wrong data and throw a missing index notice.